File: 3195-separate-black-and-white-balls/separate-black-and-white-balls.cpp
class Solution {
public:
    long long minimumSteps(string s) {
        long long n = s.size();
        long long ans = 0;

        long long cnt0 = 0;
        for(long long i = n - 1; i >= 0; --i) {
            if(s[i] == '0') {
                cnt0 ++;
            }else {
                ans += cnt0;   
            }
        }

        return ans;
    }
};

File: 3195-separate-black-and-white-balls/separate-black-and-white-balls.go
func minimumSteps(s string) int64 {
    n := len(s)
    var ans int64 = 0

    cnt0 := 0

    for i := n - 1; i >= 0; i-- {
        if s[i] == '0' {
            cnt0++
        } else {
            ans += int64(cnt0)
        }
    }

    return ans
}
